### PR TITLE
[CR]Partially jsonize body parts

### PR DESCRIPTION
**Pull request by Coolthulhu**
I moved body part data to json. This does not allow defining new body parts yet, but is a necessary change if we ever want more generic body part system.

[CR] because I need someone to comment on translations here:
* Name and encumbrance text can be translated just fine
* Accusative name is always identical to name, but requires context-sensitive gettext function
* Heading name has singular and plural variant. Are there any supported languages that have more than those two?

I play IVAN too much recently.

### CODE REVIEW

File: src/bodypart.cpp [hunk 1]
```diff
@@ -2,13 +2,57 @@
...
+template<>
+side string_to_enum<side>( const std::string &data )
+{
+    return string_to_enum_look_up( side_map, data );
+}
+
+}
+
+static std::array < body_part_struct, num_bp + 1 > human_anatomy;
+static std::unordered_map<std::string, body_part> body_parts_map;
```
Comment (codemime) [inline src/bodypart.cpp]: Since you've included `generic_factory`, why don't you use it instead?

File: data/json/body_parts.json [hunk 2]
```diff
@@ -0,0 +1,171 @@
+[
+  {
+    "id": "torso",
+    "type": "body_part",
```
Comment (codemime) [inline data/json/body_parts.json]: Utterly minor: `type` usually goes first.

Comment (Coolthulhu) [inline data/json/body_parts.json]: `format.conf` puts `id` before `type`. Files where this isn't the case are most likely not linted.
I agree that `type` being first makes more sense, though.

File: src/bodypart.cpp (hunk 1 above)
Comment (Coolthulhu) [inline src/bodypart.cpp]: Doesn't it require a proper string id to work? At the moment the "id" of body part is an enum, so I'd rather implement the generic factory parts once I can get rid of the fake IDs.

**Comment by codemime** [comment]
Has a merge conflict, looks good otherwise.

**Comment by Coolthulhu** [comment]
Should be fixed now

**Pull request merged**